locust/runners.py

```diff
@@ -567,19 +637,30 @@ def start(self, user_count, spawn_rate):
             )
             return
 
+        for user_class in self.user_classes:
+            if self.environment.host is not None:
+                user_class.host = self.environment.host
+
+        self.target_user_classes_count = weight_users(self.user_classes, user_count)
+
         self.spawn_rate = spawn_rate
-        worker_num_users = user_count // (num_workers or 1)
-        worker_spawn_rate = float(spawn_rate) / (num_workers or 1)
-        remaining = user_count % num_workers
 
         logger.info(
-            "Sending spawn jobs of %d users and %.2f spawn rate to %d ready clients"
-            % (worker_num_users, worker_spawn_rate, num_workers)
+            "Sending spawn jobs of %d users at %.2f spawn rate to %d ready clients"
+            % (user_count, spawn_rate, num_workers)
         )
 
-        if worker_spawn_rate > 100:
+        # Prior to the refactoring from https://github.com/locustio/locust/pull/1621, this warning
+        # was logged if `spawn_rate / number_of_workers` was above 100. However, the master
+        # is now responsible for dispatching and controlling the spawn rate which is more CPU intensive for
+        # the master. The number 100 is a little arbitrary as the computational load on the master greatly
+        # depends on the number of workers and the number of user classes. For instance, 5 user classes and 5 workers
+        # can easily do 200/s. However, 200/s with 50 workers and 20 user classes will likely make the dispatch very
+        # slow because of the required computations. I (@mboutet) doubt that many Locust's users are spawning
+        # that rapidly. If so, then they'll likely open issues on GitHub in which case I'll (@mboutet) take a look.
+        if spawn_rate > 100:
```

Comment:
Sounds like you are worried about a completely different issue (high load on master) than what this code was trying to warn about (high load on workers due to lots of Users/HTTP sessions being started at the same time)? 

I added this warning because people (actually mostly people who had misunderstood the load model) kept filing bugs/SO issues about poor performance on workers during fast ramp-up, so I'd like to keep that. If you want to add a separate warning about high _total_ spawn rate that is fine.